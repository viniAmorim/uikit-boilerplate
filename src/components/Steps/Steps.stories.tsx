import React from 'react';

import { Story, Meta } from '@storybook/react';

import Steps, { StepsProps } from './Steps';

export default {
  title: 'Atoms/Steps',
  component: Steps,
} as Meta;

const Template: Story<StepsProps> = (args) => <Steps {...args} />;

export const Default = Template.bind({});
Default.args = {
  titleOne: 'test',
  titleTwo: 'test',
  titleThree: 'test',
};

export const First = Template.bind({});
First.args = {
  stepOne: true,
  titleOne: 'test',
  titleTwo: 'test',
  titleThree: 'test',
};

export const Second = Template.bind({});
Second.args = {
  stepOne: true,
  stepTwo: true,
  titleOne: 'test',
  titleTwo: 'test',
  titleThree: 'test',
};

export const Third = Template.bind({});
Third.args = {
  stepOne: true,
  stepTwo: true,
  stepThree: true,
  titleOne: 'test',
  titleTwo: 'test',
  titleThree: 'test',
};
