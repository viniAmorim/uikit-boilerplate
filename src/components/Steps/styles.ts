import styled, { css } from 'styled-components';

export type StyledDotProps = {
  variant: 'active' | 'default';
};

const variantModifier = {
  default: css`
    background: ${({ theme }) => theme.colors.grey500};
  `,
  active: css`
    background: ${({ theme }) => theme.colors.pink500};
  `,
};

export const Container = styled.div`
  display: flex;
  flex-direction: row;
  width: 300px;
  margin-bottom: 40px;
  margin-top: 60px;
  justify-content: center;
  align-items: center;

  @media (min-width: 625px) {
    width: 500px;
  }
`;

export const Title = styled.p`
  font-family: 'Montserrat';
  font-style: normal;
  font-weight: 400;
  font-size: 16px;
  line-height: 26px;
`;

export const Dot = styled.div<StyledDotProps>`
  width: 24px;
  height: 24px;
  border-radius: 50%;
  ${({ variant = 'active' }) => variantModifier[variant]}
`;

export const Bar = styled.div`
  width: 100%;
  height: 5px;
  background: ${({ theme }) => theme.colors.grey100};
`;

export const Box = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 24px;
`;

export const BarBox = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: end;
  align-items: flex-end;
  height: 64px;
  width: 100%;
`;
