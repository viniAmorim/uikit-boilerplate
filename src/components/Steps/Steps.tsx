import React from 'react';

import * as Styled from './styles';

export interface StepsProps {
  stepOne?: boolean;
  stepTwo?: boolean;
  stepThree?: boolean;
  titleOne: string;
  titleTwo: string;
  titleThree: string;
}

const Steps = ({
  stepOne,
  stepTwo,
  stepThree,
  titleOne,
  titleTwo,
  titleThree,
}: StepsProps) => {
  return (
    <Styled.Container>
      <Styled.Box>
        <Styled.Title>{titleOne}</Styled.Title>
        <Styled.Dot variant={stepOne ? 'active' : 'default'} />
      </Styled.Box>

      <Styled.BarBox>
        <Styled.Bar />
      </Styled.BarBox>

      <Styled.Box>
        <Styled.Title>{titleTwo}</Styled.Title>
        <Styled.Dot variant={stepTwo ? 'active' : 'default'} />
      </Styled.Box>

      <Styled.BarBox>
        <Styled.Bar />
      </Styled.BarBox>

      <Styled.Box>
        <Styled.Title>{titleThree}</Styled.Title>
        <Styled.Dot variant={stepThree ? 'active' : 'default'} />
      </Styled.Box>
    </Styled.Container>
  );
};

export default Steps;
