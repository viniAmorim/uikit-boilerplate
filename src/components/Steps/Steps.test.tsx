import React from 'react';

import snapshotDiff from 'snapshot-diff';

import { render } from '~/test/utils';

import Steps from './Steps';

describe('Steps', () => {
  it('should render correctly', () => {
    const { container } = render(
      <Steps titleOne="test" titleTwo="test" titleThree="test" />,
    );
    const steps = container.firstChild as HTMLElement;

    expect(steps).toBeInTheDocument();
    expect(steps).toMatchSnapshot();
  });

  it('can all balls be active', () => {
    const { rerender, asFragment } = render(
      <Steps titleOne="test" titleTwo="test" titleThree="test" />,
    );
    const initial = asFragment();

    rerender(
      <Steps stepOne titleOne="test" titleTwo="test" titleThree="test" />,
    );
    const first = asFragment();

    rerender(
      <Steps
        stepOne
        stepTwo
        titleOne="test"
        titleTwo="test"
        titleThree="test"
      />,
    );
    const second = asFragment();

    rerender(
      <Steps
        stepOne
        stepTwo
        stepThree
        titleOne="test"
        titleTwo="test"
        titleThree="test"
      />,
    );
    const third = asFragment();

    expect(snapshotDiff(initial, first)).toMatchSnapshot();
    expect(snapshotDiff(initial, second)).toMatchSnapshot();
    expect(snapshotDiff(initial, third)).toMatchSnapshot();

    expect(snapshotDiff(first, second)).toMatchSnapshot();
    expect(snapshotDiff(first, third)).toMatchSnapshot();

    expect(snapshotDiff(second, third)).toMatchSnapshot();
  });
});
