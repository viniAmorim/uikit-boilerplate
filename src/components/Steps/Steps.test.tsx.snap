// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Steps can all balls be active 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -9,11 +9,11 @@
          class=\\"sc-gueYoa eGxfdf\\"
        >
          test
        </p>
        <div
-         class=\\"sc-dmqHEX blAMYx\\"
+         class=\\"sc-dmqHEX eDFYpJ\\"
        />
      </div>
      <div
        class=\\"sc-jTrPJq jhXSGZ\\"
      >"
`;

exports[`Steps can all balls be active 2`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -9,11 +9,11 @@
          class=\\"sc-gueYoa eGxfdf\\"
        >
          test
        </p>
        <div
-         class=\\"sc-dmqHEX blAMYx\\"
+         class=\\"sc-dmqHEX eDFYpJ\\"
        />
      </div>
      <div
        class=\\"sc-jTrPJq jhXSGZ\\"
      >
@@ -28,11 +28,11 @@
          class=\\"sc-gueYoa eGxfdf\\"
        >
          test
        </p>
        <div
-         class=\\"sc-dmqHEX blAMYx\\"
+         class=\\"sc-dmqHEX eDFYpJ\\"
        />
      </div>
      <div
        class=\\"sc-jTrPJq jhXSGZ\\"
      >"
`;

exports[`Steps can all balls be active 3`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -9,11 +9,11 @@
          class=\\"sc-gueYoa eGxfdf\\"
        >
          test
        </p>
        <div
-         class=\\"sc-dmqHEX blAMYx\\"
+         class=\\"sc-dmqHEX eDFYpJ\\"
        />
      </div>
      <div
        class=\\"sc-jTrPJq jhXSGZ\\"
      >
@@ -28,11 +28,11 @@
          class=\\"sc-gueYoa eGxfdf\\"
        >
          test
        </p>
        <div
-         class=\\"sc-dmqHEX blAMYx\\"
+         class=\\"sc-dmqHEX eDFYpJ\\"
        />
      </div>
      <div
        class=\\"sc-jTrPJq jhXSGZ\\"
      >
@@ -47,10 +47,10 @@
          class=\\"sc-gueYoa eGxfdf\\"
        >
          test
        </p>
        <div
-         class=\\"sc-dmqHEX blAMYx\\"
+         class=\\"sc-dmqHEX eDFYpJ\\"
        />
      </div>
    </div>
  </DocumentFragment>"
`;

exports[`Steps can all balls be active 4`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -28,11 +28,11 @@
          class=\\"sc-gueYoa eGxfdf\\"
        >
          test
        </p>
        <div
-         class=\\"sc-dmqHEX blAMYx\\"
+         class=\\"sc-dmqHEX eDFYpJ\\"
        />
      </div>
      <div
        class=\\"sc-jTrPJq jhXSGZ\\"
      >"
`;

exports[`Steps can all balls be active 5`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -28,11 +28,11 @@
          class=\\"sc-gueYoa eGxfdf\\"
        >
          test
        </p>
        <div
-         class=\\"sc-dmqHEX blAMYx\\"
+         class=\\"sc-dmqHEX eDFYpJ\\"
        />
      </div>
      <div
        class=\\"sc-jTrPJq jhXSGZ\\"
      >
@@ -47,10 +47,10 @@
          class=\\"sc-gueYoa eGxfdf\\"
        >
          test
        </p>
        <div
-         class=\\"sc-dmqHEX blAMYx\\"
+         class=\\"sc-dmqHEX eDFYpJ\\"
        />
      </div>
    </div>
  </DocumentFragment>"
`;

exports[`Steps can all balls be active 6`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -47,10 +47,10 @@
          class=\\"sc-gueYoa eGxfdf\\"
        >
          test
        </p>
        <div
-         class=\\"sc-dmqHEX blAMYx\\"
+         class=\\"sc-dmqHEX eDFYpJ\\"
        />
      </div>
    </div>
  </DocumentFragment>"
`;

exports[`Steps should render correctly 1`] = `
<div
  class="sc-beqWaB eckzXC"
>
  <div
    class="sc-eDDNvR dFFHAH"
  >
    <p
      class="sc-gueYoa eGxfdf"
    >
      test
    </p>
    <div
      class="sc-dmqHEX blAMYx"
    />
  </div>
  <div
    class="sc-jTrPJq jhXSGZ"
  >
    <div
      class="sc-hLseeU jyEoAW"
    />
  </div>
  <div
    class="sc-eDDNvR dFFHAH"
  >
    <p
      class="sc-gueYoa eGxfdf"
    >
      test
    </p>
    <div
      class="sc-dmqHEX blAMYx"
    />
  </div>
  <div
    class="sc-jTrPJq jhXSGZ"
  >
    <div
      class="sc-hLseeU jyEoAW"
    />
  </div>
  <div
    class="sc-eDDNvR dFFHAH"
  >
    <p
      class="sc-gueYoa eGxfdf"
    >
      test
    </p>
    <div
      class="sc-dmqHEX blAMYx"
    />
  </div>
</div>
`;
